File: api/recipe-ranking.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';

export default async function handler(
  req: VercelRequest,
  res: VercelResponse,
) {
  // 環境変数からIDを安全に取得
  const appId = process.env.RAKUTEN_APP_ID;
  // フロントエンドから送られてきたカテゴリIDを取得
  const categoryId = req.query.categoryId as string;

  if (!categoryId) {
    return res.status(400).json({ error: 'カテゴリが指定されていません。' });
  }
  if (!appId) {
    return res.status(500).json({ error: 'API認証情報がサーバー側で設定されていません。' });
  }

  const apiUrl = `https://app.rakuten.co.jp/services/api/Recipe/CategoryRanking/20170426?format=json&applicationId=${appId}&categoryId=${categoryId}`;

  try {
    const apiResponse = await fetch(apiUrl);
    const data = await apiResponse.json();

    if (apiResponse.ok) {
      res.status(200).json(data);
    } else {
      res.status(apiResponse.status).json(data);
    }
  } catch (error) {
    res.status(500).json({ error: 'サーバーでエラーが発生しました。' });
  }
}

File: backend/api/recipe_ranking.py
# backend/api/recipe_ranking.py

import os
import httpx
from fastapi import APIRouter, HTTPException, Query

router = APIRouter()

@router.get("/ranking")  # ✅ URLをシンプルに変更
async def recipe_ranking(categoryId: str = Query(...)):
    app_id = os.getenv("RAKUTEN_APP_ID")
    if not app_id:
        raise HTTPException(status_code=500, detail="API認証情報が設定されていません。")

    url = (
        "https://app.rakuten.co.jp/services/api/Recipe/CategoryRanking/20170426"
        f"?format=json&applicationId={app_id}&categoryId={categoryId}"
    )

    async with httpx.AsyncClient() as client:
        response = await client.get(url)

    if response.status_code == 200:
        return response.json()
    else:
        raise HTTPException(status_code=response.status_code, detail=response.text)
